src/OpenFeature.Contrib.Providers.Flipt/README.md

```diff
@@ -0,0 +1,129 @@
+# Flipt .NET Provider
+
+The flipt provider allows you to connect to your Flipt instance through the OpenFeature SDK
+
+# .Net SDK usage
+
+## Requirements
+
+- open-feature/dotnet-sdk v1.5.0 > v2.0.0
+
+## Install dependencies
+
+The first things we will do is install the **Open Feature SDK** and the **Flipt Feature Flag provider**.
+
+### .NET Cli
+
+```shell
+dotnet add package OpenFeature.Contrib.Providers.Flipt
+```
+
+### Package Manager
+
+```shell
+NuGet\Install-Package OpenFeature.Contrib.Providers.Flipt
+```
+
+### Package Reference
+
+```xml
+<PackageReference Include="OpenFeature.Contrib.Providers.Flipt" />
+```
+
+### Packet cli
+
+```shell
+packet add OpenFeature.Contrib.Providers.Flipt
+```
+
+### Cake
+
+```shell
+// Install OpenFeature.Contrib.Providers.Flipt as a Cake Addin
+#addin nuget:?package=OpenFeature.Contrib.Providers.Flipt
+
+// Install OpenFeature.Contrib.Providers.Flipt as a Cake Tool
+#tool nuget:?package=OpenFeature.Contrib.Providers.Flipt
+```
+
+## Using the Flipt Provider with the OpenFeature SDK
+
+To create a Flagmith provider you should define provider and Flipt settings.
+
+```csharp
+using OpenFeature.Contrib.Providers.Flipt;
+using OpenFeature.Model;
+
+var featureProvider = new FliptProvider("http://localhost:8080", "default-namespace");
+
+// Set the featureProvider as the provider for the OpenFeature SDK
+await OpenFeature.Api.Instance.SetProviderAsync(featureProvider);
+
+// Get an OpenFeature client
+var client = OpenFeature.Api.Instance.GetClient();
+
+// Optional: set EntityId and updated context
+var context = EvaluationContext.Builder()
+    .SetTargetingKey("flipt EntityId")
+    .Set("extra-data-1", "extra-data-1-value")
+    .Build();
+
+// Evaluate a flag
+var val = await client.GetBooleanValueAsync("myBoolFlag", false, context);
+
+// Print the value of the 'myBoolFlag' feature flag
+Console.WriteLine(val);
+```
+
+# Contribution
+
+## Code setup
+
+Since the official [flipt-csharp](https://github.com/flipt-io/flipt-server-sdks/tree/main/flipt-csharp) only supports
+dotnet 8.0, it was not utilized in this provider as OpenFeature aims to support a bigger range of dotnet versions.
+
+### Rest Client using OpenAPI
+
+To work around this incompatibility, the openapi specification of [Flipt](https://github.com/flipt-io/flipt-openapi) was
+used to generate a REST client using [nswag](https://github.com/RicoSuter/NSwag).
+
+## Updating the REST Client
+
+To generate or update the Flipt REST client, follow these steps:
+
+### 1. Download the OpenAPI Specification
+
+First, download the latest `openapi.yml` file from the Flipt GitHub repository. This can be done manually or by using a
+command like `curl`:
+
+```
+
+curl https://raw.githubusercontent.com/flipt-io/flipt-openapi/main/openapi.yml -o openapi.yml
```

Comment:
this has actually moved here: https://github.com/flipt-io/flipt/blob/main/openapi.yaml

The previous open-api repo has been archived. I need to add a note to the readme, thanks for the reminder!